kfctl/README.md

```diff
@@ -0,0 +1,140 @@
+# kfctl golang client
+
+## Overview
+
+The kfctl golang client will provide a CLI similar to kfctl.sh but will 
+address the following deficits in the current tool
+
+1. No longer require a local install of ks (ksonnet)
+2. No longer require knowledge of the ks CLI 
+3. No longer save a ksonnet application to the user's filesystem
+4. Work in a similar way as the kubeflow UI [gcp-click-to-deploy](https://github.com/kubeflow/kubeflow/tree/master/components/gcp-click-to-deploy):
+  - Execution of subcommands `init`, `generate`, `apply` and `delete` are done by the bootstrapper.
+  - A local `<kf_app>.yaml` defines a kubeflow application. 
+  - This yaml file is submitted to the bootstrapper which will then generate the ksonnet application.
+
+## Usage
+
+The initial version of kfctl will seek parity with kfctl.sh by implementing the following subcommands:
+- `init`            Initialize a kubeflow application.
+- `generate`        Generate the k8 manifests of the kubeflow application.
+- `apply`           Submit the k8 manifests to the api-server
+- `delete`          Delete the kubeflow application
+
+The current usage of `kfctl.sh` is as follows:
+
+```sh
+kfctl.sh init myapp --platform generatic
+cd myapp
+kfctl.sh generate all
+kfctl.sh apply all
+```
+
+kfctl (golang) usage is similar:
+
+```sh
+kfctl init myapp 
+# edit myapp.yaml to set the appAddress and optionally add components and/or parameters
```

Comment:
Why are you introducing a myapp.yaml file here?

I thought the whole purpose of the CLI was to pretty much keep the existing semantics of kfctl.sh but just reuse the go library.